src/eval_context.rs

```diff
@@ -196,7 +194,7 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
             Struct(_)    => unimplemented!(),
             Tuple(_)     => unimplemented!(),
-            Function(_)  => unimplemented!(),
+            Function(_, _)  => unimplemented!(),
```

Comment:
Oh... you mean changing it that `Literal::Item` stops referring to functions. Sounds good.